### ISSUE TITLE
Changing the name of an agent in the DSL does not delete the old agent

### ISSUE THREAD

**Issue by kaikreuzer** [lmos-ai/arc#29]
Using Arc runner and using hot-deploy, here's how to reproduce:

1. Create an agent
2. Rename the agent in the DSL (keep the filename unchanged)
3. Refresh the browser
4. The old and the new name will appear as agents.

**Comment by patwlan** [comment]
Correct, we will need to do some conceptual work on "removing" an agent. There may be some clean process up required.
